src/rawBinaryConn.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * This source file is part of the EdgeDB open source project.
+ *
+ * Copyright 2019-present MagicStack Inc. and the EdgeDB authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {NULL_CODEC} from "./codecs/codecs";
+import {CodecsRegistry} from "./codecs/registry";
+import {EMPTY_TUPLE_CODEC} from "./codecs/tuple";
+import {NormalizedConnectConfig} from "./conUtils";
+import {PrepareMessageHeaders, ProtocolVersion} from "./ifaces";
+import {WriteBuffer} from "./primitives/buffer";
+import {RawConnection} from "./rawConn";
+import {retryingConnect} from "./retry";
+import {versionGreaterThanOrEqual} from "./utils";
+
+export class RawBinaryConnection {
```

Comment:
I don't think we need this type -- let's just add the necessary "public" methods directly to `RawConnection` and export it? One less class to care about.

Reply:
Done in https://github.com/edgedb/edgedb-js/pull/203/commits/88aeb6168365759e97313584c2ef04de37ec1687